### ISSUE TITLE
Reduce buildbox image size

### ISSUE THREAD

**Issue by awly** [gravitational/teleport#6149]
The current buildbox image is >1GB.
This hits our CI performance and cost.

We should trim the amount of data in the image and/or use multi-stage builds.

**Comment by webvictim** [comment]
I'm working on https://github.com/gravitational/teleport/pull/6145 as a quick resolution to Docker-related slowdowns as well.

I've also just realised that Drone's sqlite database is 4GB in size and located on a `standard` GKE PV which uses spinning disks rather than SSDs.

**Comment by gliptak** [comment]
are the sources for https://quay.io/repository/gravitational/teleport-buildbox?tag=latest&tab=tags public?

the up-to-date release has a number of vulnerabilities identified https://quay.io/repository/gravitational/teleport-buildbox/manifest/sha256:0a493df70dbdc593b84df9de8eb740e178838afde3123c5c1b01a539186d1825?tab=vulnerabilities

**Comment by webvictim** [comment]
Yes - https://github.com/gravitational/teleport/blob/master/build.assets/Dockerfile

**Comment by tcsc** [comment]
Some candidates for easy wins: 

- Delete the `gcloud` backup directory `/opt/google-cloud-sdk/.install/.backup` after installing gcloud sdk & firestore emulator
- Make a shallow clone of the protobuf repo
- Delete the `etcd` tarball after it's unpacked
- Run `make clean` after `make install` for protobuf & libbpf

**Comment by tcsc** [comment]
> The current buildbox image is >1GB.

This seems so quaint now, as our buildbox is currently > 6GB

**Comment by zmb3** [comment]
> This seems so quaint now, as our buildbox is currently > 6GB

Hey, we're down to under 4.5GB now :-)
